src/imap.rs

```diff
@@ -1595,26 +1524,29 @@ impl Imap {
         if uid == 0 {
             return Some(ImapActionResult::RetryLater);
         }
-        if self.session.is_none() {
-            // currently jobs are only performed on the INBOX thread
-            // TODO: make INBOX/SENT/MVBOX perform the jobs on their
-            // respective folders to avoid select_folder network traffic
-            // and the involved error states
-            if let Err(err) = self.prepare(context).await {
-                warn!(context, "prepare_imap_op failed: {}", err);
-                return Some(ImapActionResult::RetryLater);
-            }
+        if let Err(err) = self.prepare(context).await {
+            warn!(context, "prepare_imap_op failed: {}", err);
+            return Some(ImapActionResult::RetryLater);
         }
-        match self.select_folder(context, Some(folder)).await {
+
+        let session = match self
+            .session
+            .as_mut()
+            .context("no IMAP connection established")
```

Comment:
Do we need this if we have `error!(...)` three lines below?

Reply:
Ok, i just meant we can write all we need inside that `error!`. It doesn't matter much